src/wam2layers/analysis/checks.py

```diff
@@ -10,7 +10,7 @@ def _warning_on_one_line(message, category, filename, lineno, file=None, line=No
 
     https://stackoverflow.com/a/26433913
     """
-    short_path = Path(filename).relative_to(Path(filename).parents[2])
+    short_path = Path(filename).relative_to(Path(filename).parents[0])
```

Comment:
Change made here to fix the following error:

```py
❯ pytest tests/test_workflow.py
================================================= test session starts ==================================================
platform linux -- Python 3.8.10, pytest-7.4.2, pluggy-1.3.0
rootdir: /home/yangliu/WAM2layers/WAM2layers
collected 2 items

tests/test_workflow.py .
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/_pytest/main.py", line 271, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/_pytest/main.py", line 325, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_callers.py", line 152, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_result.py", line 114, in get_result
INTERNALERROR>     raise exc.with_traceback(exc.__traceback__)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/_pytest/main.py", line 350, in pytest_runtestloop
INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_hooks.py", line 493, in __call__
INTERNALERROR>     return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_callers.py", line 130, in _multicall
INTERNALERROR>     teardown[0].send(outcome)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/_pytest/warnings.py", line 109, in pytest_runtest_protocol
INTERNALERROR>     yield
INTERNALERROR>   File "/usr/lib/python3.8/contextlib.py", line 120, in __exit__
INTERNALERROR>     next(self.gen)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/_pytest/warnings.py", line 64, in catch_warnings_for_item
INTERNALERROR>     ihook.pytest_warning_recorded.call_historic(
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_hooks.py", line 514, in call_historic
INTERNALERROR>     res = self._hookexec(self.name, self._hookimpls, kwargs, False)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_manager.py", line 115, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_callers.py", line 113, in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/pluggy/_callers.py", line 77, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/_pytest/terminal.py", line 538, in pytest_warning_recorded
INTERNALERROR>     message = warning_record_to_str(warning_message)
INTERNALERROR>   File "/home/yangliu/venv/wam2layers/lib/python3.8/site-packages/_pytest/warnings.py", line 77, in warning_record_to_str
INTERNALERROR>     msg = warnings.formatwarning(
INTERNALERROR>   File "/home/yangliu/WAM2layers/WAM2layers/src/wam2layers/analysis/checks.py", line 13, in _warning_on_one_line
INTERNALERROR>     short_path = Path(filename).relative_to(Path(filename).parents[2])
INTERNALERROR>   File "/usr/lib/python3.8/pathlib.py", line 621, in __getitem__
INTERNALERROR>     raise IndexError(idx)
INTERNALERROR> IndexError: 2
```

I think the this function was created to make the warning message a bit shorter. However, it triggers the error above when running the test. Let me know if there is a better solution, other than changing the depth.

Reply:
Eventually I think we need to use these checks in another way, but I guess this is OK for now.